exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/AbstractParquetScanBatchCreator.java

```diff
@@ -161,20 +167,24 @@ private ParquetMetadata readFooter(Configuration conf, String path) throws IOExc
     }
   }
 
-  private boolean isComplex(ParquetMetadata footer) {
+  private boolean isComplex(ParquetMetadata footer, List<SchemaPath> columns) {
     MessageType schema = footer.getFileMetaData().getSchema();
 
-    for (Type type : schema.getFields()) {
-      if (!type.isPrimitive()) {
-        return true;
+    if (Utilities.isStarQuery(columns)) {
+      for (Type type : schema.getFields()) {
+        if (!type.isPrimitive()) {
+          return true;
+        }
       }
-    }
-    for (ColumnDescriptor col : schema.getColumns()) {
-      if (col.getMaxRepetitionLevel() > 0) {
-        return true;
+      for (ColumnDescriptor col : schema.getColumns()) {
+        if (col.getMaxRepetitionLevel() > 0) {
+          return true;
+        }
       }
+      return false;
+    } else {
+      return containsComplexColumn(footer, columns);
```

Comment:
Basically `isComplex` and `containsComplexColumn` does the same, why would split it into semantically the same methods?

Reply:
I based this one on original PR, but indeed it's a valid point. Will regroup accordingly.

Reply:
Methods combined in one and moved to utility class.